core/mainstage.go

```diff
@@ -398,5 +406,13 @@ func (sm *stages) mainHandleEvent(e events.Event) {
 		if e.IsHandled() || st.Modal || st.Type == WindowStage || st.FullWindow {
 			break
 		}
+		if st.Type == DialogStage { // modal dialog, by definition
+			if e.HasPos() {
+				b := st.Scene.SceneGeom.Bounds()
+				if e.WindowPos().In(b) { // don't propagate
+					break
+				}
+			}
+		}
```

Comment:
This is what I am talking about. Shouldn't this logic prevent the events from escaping from the modeless dialog? By the way, your comment is wrong, this is `modeless dialog, by definition`. And is `WindowPos` correct here (vs `Pos`)?